### PR TITLE
[Tcp] Add bazel rules for targets and tests in torch-mlir-dialects

### PR DESCRIPTION
**Pull request by navahgar**
This PR includes the following:
 * Fixes an issue with `TorchMLIRTcpToLinalg` target.
 * Adds bazel rules for targets in `torch-mlir-dialects` - `TorchMLIRDialectsConversionPasses` and `torch-mlir-dialects-opt`
 * Updates `TorchMLIRInitAll` target to include `Tcp` dialect and `TorchMLIRDialectsConversionPasses`
 * Adds bazel rules for lit tests in Tcp dialect and TcpToLinalg conversion, as well as the lit tests for TMTensor dialect.
 * Updated GHA workflow for Bazel Build and Test to build and test torch-mlir-dialects.
 * Added a passing test to the TCP pass list in e2e testing.

GHA run for Bazel build and test: https://github.com/navahgar/torch-mlir/actions/runs/3851430260/workflow

### CODE REVIEW

File: utils/bazel/torch-mlir-overlay/externals/llvm-external-projects/torch-mlir-dialects/test/Conversion/BUILD.bazel [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# This file is licensed under the Apache License v2.0 with LLVM Exceptions.
```
Comment (sjain-stanford) [inline utils/bazel/torch-mlir-overlay/externals/llvm-external-projects/torch-mlir-dialects/test/Conversion/BUILD.bazel:1]: For completeness, could you also enable the LIT tests in the `tmtensor` directory?

**Review by sjain-stanford — APPROVED** [review]
LGTM with two minor comments.

File: utils/bazel/torch-mlir-overlay/externals/llvm-external-projects/torch-mlir-dialects/test/BUILD.bazel [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    ],
+)
+
+expand_template(
+    name = "lit_site_cfg_py",
+    testonly = True,
+    out = "lit.site.cfg.py",
+    substitutions = {
+        "@LIT_SITE_CFG_IN_HEADER@": "# Autogenerated, do not edit.",
+        "@TORCH_MLIR_SOURCE_DIR@": package_path("@torch-mlir//:BUILD"),
```
[the remark this answers is not in the document]
Comment (navahgar) [reply utils/bazel/torch-mlir-overlay/externals/llvm-external-projects/torch-mlir-dialects/test/BUILD.bazel]: Done.

File: utils/bazel/torch-mlir-overlay/externals/llvm-external-projects/torch-mlir-dialects/test/Conversion/BUILD.bazel (hunk 1 above)
Comment (navahgar) [reply utils/bazel/torch-mlir-overlay/externals/llvm-external-projects/torch-mlir-dialects/test/Conversion/BUILD.bazel:1]: Sure, done.

**Review by sjain-stanford — APPROVED** [review]
LGTM!

**Pull request merged**